src/llamafactory/train/trainer_utils.py

```diff
@@ -371,6 +371,12 @@ def _create_badam_optimizer(
         dict(params=decay_params, weight_decay=training_args.weight_decay),
     ]
 
+    ds_zero3_enabled = False
+    if hasattr(training_args, "deepspeed_plugin") and training_args.deepspeed_plugin is not None:
```

Comment:
Why not use `from transformers.integrations import is_deepspeed_zero3_enabled`

Reply:
Thanks for the suggestion, I have changed it to use `is_deepspeed_zero3_enabled` for cleaner expressions.